scripts/gce_integration_test.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# - Builds the test app;
+# - Creates a GCE VM as per
+#   https://cloud.google.com/container-optimized-os/docs/how-to/create-configure-instance.
+
+# Immediately exits in the presence of an error.
+set -e
+
+readonly dir=$(dirname $0)
+readonly projectRoot="$dir/.."
+readonly testAppDir="$projectRoot/test-application"
+readonly deployDir="$testAppDir/target/deploy"
+readonly imageName="openjdk-gce-integration-test"
+
+readonly imageUnderTest=$1
+if [[ -z "$imageUnderTest" ]]; then
+  echo "Usage: ${0} <image_under_test>"
+  exit 1
+fi
+
+readonly projectName=$(gcloud info \
+                | awk '/^Project: / { print $2 }' \
+                | sed 's/\[//'  \
+                | sed 's/\]//')
+
+readonly imageUrl="gcr.io/$projectName/$imageName"
+
+# Build the test app.
+pushd ${testAppDir}
+mvn clean package
+popd
+
+pushd ${deployDir}
+export STAGING_IMAGE=${imageUnderTest}
+envsubst < "Dockerfile.in" > "Dockerfile"
+export TESTED_IMAGE=${imageUrl}
+envsubst < "openjdk-spring-boot.yaml.in" > "openjdk-spring-boot.yaml"
+
+echo "Deploying image to Google Container Registry..."
+gcloud docker -- build -t "$imageName" .
```

Comment:
The gcloud docker command is necessary in case you use a base image which is located on a private Container Registry (which is the default behavior of build.sh).